python/ray/data/datasource/parquet_datasource.py

```diff
@@ -243,69 +180,138 @@ def read_pieces(serialized_pieces: str) -> Iterator[pa.Table]:
                 inferred_schema = schema
         else:
             inferred_schema = schema
+        self._metadata = meta_provider.prefetch_file_metadata(pq_ds.pieces) or []
+        self._pq_ds = pq_ds
+        self._meta_provider = meta_provider
+        self._inferred_schema = inferred_schema
+        self._block_udf = _block_udf
+        self._reader_args = reader_args
+        self._columns = columns
+        self._schema = schema
+
+    def estimate_inmemory_data_size(self) -> Optional[int]:
```

Comment:
sorry for the late comment. but I think it's probably a bug here to rely on `serialized_size`, which seems to be size of file footer, not size of actual data. Crafted a quick fix in https://github.com/ray-project/ray/pull/26516 , please let me know if it makes sense or not, thanks.